tests/jobs/scala/src/main/scala/KafkaCassandra.scala

```diff
@@ -0,0 +1,262 @@
+      System.exit(1)
+    }
+
+    val config = parser.get
+    println(s"Using config: $config")
+
+    val spark = StreamingExamples.createSparkSession(config)
+
+    val streamingContext = new StreamingContext(spark.sparkContext, Seconds(config.batchSizeSeconds))
+
+    // Create direct kafka stream with brokers and topics
+    val topicsSet = config.topics.split(",").toSet
+
+    val kafkaParams = getKafkaProperties(config.brokers, config.groupId, config.kerberized)
+
+    val cassandraEnabled : Boolean = config.cassandra
+
+    // Set up the cassandra session:
+    val keyspaceName = config.cassandraKeyspace
+    val tableName = config.cassandraTable
+    val cassandraColumns = SomeColumns("ts" as "_1", "word" as "_2", "count" as "_3")
+
+    if (cassandraEnabled) {
+      CassandraConnector(spark.sparkContext.getConf).withSessionDo { session =>
+
+        session.execute(
+          s"""CREATE KEYSPACE IF NOT EXISTS $keyspaceName WITH REPLICATION = {
+            'class': 'SimpleStrategy',
+            'replication_factor': 3
+          }""")
+        session.execute(
+          s"""
+            CREATE TABLE IF NOT EXISTS $keyspaceName.$tableName (
+              ts timestamp,
+              word text,
+              count int,
+              PRIMARY KEY(word, ts)
+            );"""
+        )
+      }
+    }
+
+    val messages = KafkaUtils.createDirectStream[String, String](
+      streamingContext,
+      LocationStrategies.PreferConsistent,
+      ConsumerStrategies.Subscribe[String, String](topicsSet, kafkaParams))
+
+    // Get the lines, split them into words, count the words and print
+    val lines = messages.map(_.value)
+    val words = lines.flatMap(_.split(" "))
+    val wordCounts = words
+      .map(x => (x, 1L))
+      .reduceByKey(_ + _)
+      .map(x => (new Date().getTime(), x._1, x._2))
+
+    if (cassandraEnabled) {
+      wordCounts.foreachRDD(rdd => {
+        println(s"Writing ${rdd.toString} to Cassandra")
+        rdd.saveToCassandra(keyspaceName, tableName, cassandraColumns)
+      })
+    }
+
+    wordCounts.print()
+
+    // Start the computation
+    streamingContext.start()
+    streamingContext.awaitTermination()
+  }
+
+  def getKafkaProperties(brokers : String, groupId: String, kerberized: Boolean) : Map[String, Object] = {
+
+    val props = Map[String, Object](
+      ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG -> brokers,
+      ConsumerConfig.GROUP_ID_CONFIG -> groupId,
+      ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG -> classOf[StringDeserializer],
+      ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG -> classOf[StringDeserializer])
+
+    //    if (kerberized) {
+    //      props.put("sasl.kerberos.service.name", "kafka")
+    //      props.put("security.protocol", "SASL_PLAINTEXT")
+    //      props.put("sasl.mechanism", "GSSAPI")
+    //    }
+
+    props
+  }
+
+}
+
+
+/** Utility functions for Spark Streaming examples. */
+object StreamingExamples extends Logging {
```

Comment:
At this point you've got a case class and 3 separate objects in the same file. You may want to start splitting all these out: [as written here](https://docs.scala-lang.org/style/files.html)